src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -791,4 +733,102 @@ public String toString() {
             return "Key(" + baseName + ", " + locale + ")";
         }
     }
+
+    /**
+     * Registry of the loaded <code>resource bundles</code> and the associated <code>service registrations</code>
+     * The <code>ResourceBundleRegistry</code> takes care of the registration/deregistration of the resource bundles as OSGi services.
+     * It stores the references to the registered resource bundles and to the associated service registrations.
+     */
+    private static class ResourceBundleRegistry {
```

Comment:
Please implement Auto-/Closeable to clarify that this is supposed to be classed by callers.

Reply:
Thanks for the suggestion! I hesitate: in this particular case there will never be a try-with use case since the usage of ResourceBundleRegistry always extends over the length of a method (instantiate in "activate", close in "deactivate" method). 

But if you feel it would clarify that the registry must be closed at some time, I am happy to implement AutoCloseable. (Closeable throws IOException so it doesn't seem adequate). What do you think?

Reply:
I decided to implement AutoCloseable as you suggested, to empasize that the registry should be closed :-)